lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -116,6 +125,13 @@ class DatasetStorageDetails(Model):
     )
     hashes: List[dict] = Field(description="The file contents hashes associated with the supplied dataset instance.")
     sources: List[dict] = Field(description="The file sources associated with the supplied dataset instance.")
+    shareable: bool = Field(
+        description="Is this dataset shareable.",
+    )
+    quota: dict = Field(description="Information about quota sources around dataset storage.")
+    badges: List[Dict[str, Any]] = Field(
+        description="Fast summary parsed for digging into about target object store dataset is stored on."
```

Comment:
```suggestion
        description="A mapping of object store labels to badges describing object store properties."
```
? I don't know though what Any is